src/driver/DriverUtils.ts

```diff
@@ -50,21 +51,20 @@ export class DriverUtils {
      * Extracts connection data from the connection url.
      */
     private static parseConnectionUrl(url: string) {
-        const firstSlashes = url.indexOf("//");
-        const preBase = url.substr(firstSlashes + 2);
-        const secondSlash = preBase.indexOf("/");
-        const base = (secondSlash !== -1) ? preBase.substr(0, secondSlash) : preBase;
-        const afterBase = (secondSlash !== -1) ? preBase.substr(secondSlash) + 1 : undefined;
-        const [usernameAndPassword, hostAndPort] = base.split("@");
-        const [username, password] = usernameAndPassword.split(":");
-        const [host, port] = hostAndPort.split(":");
+        const {
+            hostname,
+            port = undefined,
+            auth = ":",
+            path
+        } = Url.parse(url);
+        const [username, password] = auth.split(":");
 
         return {
-            host: host,
+            host: hostname,
             username: username,
             password: password,
             port: port ? parseInt(port) : undefined,
-            database: afterBase ? afterBase.split("/")[0] : undefined
+            database: path ? path.substr(1) : undefined
         };
```

Comment:
you should revert using url package. Previously I used url package, however I decided not to use it anymore and replaced with my own code. The reason is that typeorm can be used in the browser and we should avoid using third-party dependencies. Im sure what you did - same can be achieved with previous code + bugfix